client/client-multi/client-cpp/impl/security/RsaEncoderDecoder.cpp

```diff
@@ -51,13 +52,14 @@ RsaEncoderDecoder::RsaEncoderDecoder(const PublicKey& pubKey,
     }
 
     KAA_LOG_TRACE(boost::format("RemotePublicKey: %1%") % ( remoteKey_ ? LoggingUtils::ByteArrayToString(
-            remoteKey_->x509_subject_public_key().begin(), remoteKey_->x509_subject_public_key().size()) : "empty"));
+            remoteKey_->x509_subject_public_key().data(), remoteKey_->x509_subject_public_key().size()) : "empty"));
 }
 
-Botan::SecureVector<std::uint8_t> RsaEncoderDecoder::getEncodedSessionKey()
+EncodedSessionKey RsaEncoderDecoder::getEncodedSessionKey()
 {
     Botan::PK_Encryptor_EME enc(*remoteKey_, "EME-PKCS1-v1_5");
-    return enc.encrypt(sessionKey_.bits_of(), rng_);
+    const auto &v = enc.encrypt(sessionKey_.bits_of(), rng_);
```

Comment:
Non-const references can't be initialized from rvalues. However, in C++11 it is possible to assign rvalue to non-const reference with move semantics. (T&&) - going to change this.